wallet-cli/src/command/pretty.rs

```diff
@@ -3,32 +3,45 @@ use std::fmt;
 use std::string::String;
 
 use blockchain::{genesis, normal, types, Block, SscProof};
-use wallet_crypto::{address, hash, tx, util::hex};
+use wallet_crypto::{address, cbor, config, hash, hdwallet, redeem, tx, util::hex};
 
 use ansi_term::Colour;
 
 // Constants for the fmt::Display instance
 static DISPLAY_INDENT_SIZE: usize = 4; // spaces
 static DISPLAY_INDENT_LEVEL: usize = 0; // beginning starts at zero
 
-type AST = Vec<(Key, Val)>;
+type AST<'a> = Vec<(Key<'a>, Val<'a>)>;
 
-type Key = String;
+type Key<'a> = &'a str;
 
 // XXX: consider splitting into two mutually-recursive types (one with only terminals, one with only nonterminals)
 // TODO: extend with blockchain-specific constructors with color
-pub enum Val {
+pub enum Val<'a> {
     // terminals
     Raw(String),
-    Hash(types::HeaderHash),
+    Hash(types::HeaderHash), // XXX: consider naming this with a more specific meaning, as we'll probably have other hashes?
+    //// numbers
     Epoch(u32),
     SlotId(u32),
```

Comment:
Just noticed these 2: Clearly here we want an `BlockDate` instead

Reply:
In the structure of the block we don't always get an `Epoch` together with a `SlotId`, so I think these need to stay separate.

Reply:
that's why I was talking about `BlockDate`, it is an enum that handles the Genesis and the Normal blocks separately.

Reply:
Ah, I see.. I'll make that change